src/index.ts

```diff
@@ -1,117 +1,134 @@
 import path from 'path'
-import {createCompiler} from '@mdx-js/mdx'
-import {babel as rollupBabel} from '@rollup/plugin-babel'
-import commonjs from '@rollup/plugin-commonjs'
-import json from '@rollup/plugin-json'
-import {nodeResolve} from '@rollup/plugin-node-resolve'
-import babelPresetReact from '@babel/preset-react'
-import babelPresetEnv from '@babel/preset-env'
-import babelPresetTypeScript from '@babel/preset-typescript'
+import {StringDecoder} from 'string_decoder'
+import {compile as compileMDX} from 'xdm'
+import xdm from 'xdm/esbuild.js'
+import remarkFrontmatter from 'remark-frontmatter'
+import {remarkMdxFrontmatter} from 'remark-mdx-frontmatter'
 import matter from 'gray-matter'
-import type {OutputOptions, RollupOptions} from 'rollup'
-import {rollup} from 'rollup'
-import {terser} from 'rollup-plugin-terser'
+import {build as bundle} from 'esbuild'
+import type {Plugin, BuildOptions, Loader} from 'esbuild'
+import nodeResolve from '@esbuild-plugins/node-resolve'
+import {globalExternals} from '@fal-works/esbuild-plugin-global-externals'
+import type {ModuleInfo} from '@fal-works/esbuild-plugin-global-externals'
+
+type ESBuildOptions = BuildOptions & {write: false}
 
 async function bundleMDX(
   mdxSource: string,
   {
     files = {},
-    remarkPlugins = [],
-    rollup: {
-      getInputOptions = (options: RollupOptions) => options,
-      getOutputOptions = (options: OutputOptions) => options,
-    } = {},
+    esbuildOptions = (options: ESBuildOptions) => options,
+    globals = {},
   }: {
     files?: Record<string, string>
-    remarkPlugins?: Array<unknown>
```

Comment:
Are we sure we don't want to allow users to supply remark plugins?

Reply:
I also would prefer to have remarkPlugins too, I'm using them right now for syntax highlighting.